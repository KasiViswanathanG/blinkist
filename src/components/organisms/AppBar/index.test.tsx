import {screen, fireEvent, getByText, render } from '@testing-library/react'
import React from 'react'
import renderer from 'react-test-renderer'
import AppBarComponent from '.'
import { BrowserRouter as Router} from 'react-router-dom';


describe('Testing the ImageComponent', () => {
    it('render ImageComponent', () => {
      const image = renderer.create(<Router><AppBarComponent/></Router>).toJSON();
      expect(image).toMatchSnapshot();
    });
    it('should call on click Explore', () => {
      const handleClick = jest.fn();
      render(<Router><AppBarComponent/></Router>)
      const buttonElement = screen.getByText('Explore')
      fireEvent.click(buttonElement)
    });
    it('should call on click Explore and Explore', () => {
      const handleClick = jest.fn();
      render(<Router><AppBarComponent/></Router>)
      const buttonElement = screen.getByText('Explore')
      fireEvent.click(buttonElement)
      const buttonElement2 = screen.getByTestId('ModalExplore')
      fireEvent.click(buttonElement2)
    });
    it('should call on click Explore and MyLibrary', () => {
      const handleClick = jest.fn();
      render(<Router><AppBarComponent/></Router>)
      const buttonElement = screen.getByText('Explore')
      fireEvent.click(buttonElement)
      const buttonElement2 = screen.getByTestId('ModalMyLibrary')
      fireEvent.click(buttonElement2)
    });
    it('should call on click Explore and Entrepreneurship', () => {
      const handleClick = jest.fn();
      render(<Router><AppBarComponent/></Router>)
      const buttonElement = screen.getByText('Explore')
      fireEvent.click(buttonElement)
      const buttonElement2 = screen.getByText('Entrepreneurship')
      fireEvent.click(buttonElement2)
    });
    it('should call on click Explore and Entrepreneurship on Hover', () => {
      const handleClick = jest.fn();
      render(<Router><AppBarComponent/></Router>)
      const buttonElement = screen.getByText('Explore')
      fireEvent.click(buttonElement)
      const buttonElement2 = screen.getByText('Entrepreneurship')
      fireEvent.mouseOver(buttonElement2)
      fireEvent.mouseLeave(buttonElement2)
    });
    it('should call on click My Library', () => {
      const handleClick = jest.fn();
      render(<Router><AppBarComponent/></Router>)
      const buttonElement = screen.getByText('My Library')
      fireEvent.click(buttonElement)
    });
})  