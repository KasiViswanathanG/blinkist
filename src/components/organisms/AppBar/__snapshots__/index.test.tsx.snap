// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the ImageComponent render ImageComponent 1`] = `
<header
  className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
  style={
    Object {
      "background": "transparent",
      "boxShadow": "none",
    }
  }
>
  <div
    className="MuiContainer-root MuiContainer-maxWidthXl css-19r6kue-MuiContainer-root"
  >
    <div
      className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1g467ij-MuiToolbar-root"
    >
      <div
        className="MuiBox-root css-b16a2w"
      >
        <img
          alt="logo"
          className="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-jqh5pz-MuiCardMedia-root"
          src="static/images/logo.png"
        />
        <span
          style={
            Object {
              "marginTop": 7,
            }
          }
        >
          <div
            onClick={[Function]}
            style={
              Object {
                "display": "inline-flex",
              }
            }
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              color="black"
              data-testid="SearchIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
              />
            </svg>
            <p
              className="MuiTypography-root MuiTypography-body2 css-1kjyx0o-MuiTypography-root"
            />
          </div>
        </span>
        <button
          className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-1sg0l7r-MuiButtonBase-root-MuiButton-root"
          data-testid={
            <ForwardRef(Typography)
              color="black"
            >
              Explore
            </ForwardRef(Typography)>
          }
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <p
            className="MuiTypography-root MuiTypography-body1 css-nnsfc0-MuiTypography-root"
          >
            Explore
          </p>
          <span
            className="MuiButton-endIcon MuiButton-iconSizeMedium css-9tj150-MuiButton-endIcon"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              color="black"
              data-testid="KeyboardArrowDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-77jh4r-MuiButtonBase-root-MuiButton-root"
          data-testid={
            <ForwardRef(Typography)
              color="black"
            >
              My Library
            </ForwardRef(Typography)>
          }
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <p
            className="MuiTypography-root MuiTypography-body1 css-nnsfc0-MuiTypography-root"
          >
            My Library
          </p>
          <span
            className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        className="MuiBox-root css-2uchni"
      >
        <button
          aria-label="Open settings"
          aria-labelledby={null}
          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-av538e-MuiButtonBase-root-MuiIconButton-root"
          data-mui-internal-clone-element={true}
          disabled={false}
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <div
            style={
              Object {
                "display": "inline-flex",
              }
            }
          >
            <div
              className="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-jp7pxi-MuiAvatar-root"
            >
              V
            </div>
            <span
              style={
                Object {
                  "marginTop": "9px",
                }
              }
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="KeyboardArrowDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
                />
              </svg>
            </span>
          </div>
          <span
            className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</header>
`;
